### DISCUSSION TITLE
When will Tempo metrics 3h limit be lifted?

### DISCUSSION THREAD

**Discussion by jasonkuhrt** [grafana/tempo#3857] [General]
Hey, I would like to know if there is a roadmap when Grafana Cloud Tempo will have the 3h window on tempo metrics lifted? Its currently a big limitation for us going all in with grafana. How are people dealing with this right now?

**Comment by joe-elliott** [comment]
This is not a great place to get answers about Grafana Cloud Traces. I would consider asking this question of Grafana Cloud support.

**Reply by jasonkuhrt** [reply]
Ok thanks for the tip, I'll try that. I know very little about the boundaries and how things are decoupled both technically and organizationally. IIUC none of your work here on Tempo is affecting this issue and instead its a deployment concern relegated only to Grafana Cloud? E.g. this issue wouldn't exist if we hosted Tempo ourselves? The docs mention metric queries being experimental, is that something under your technical direction or another team? And is the experimental aspect in anyway related to the 3h limit to your knowledge? https://grafana.com/docs/tempo/latest/traceql/metrics-queries/ I will write a support ticket though with these questions too. Thanks @joe-elliott.

**Reply by joe-elliott** [reply]
No worries. I just like to keep this repo purely about open source Tempo. I am the technical lead for the open source project as well as the Grafana Cloud Traces product so you're not wrong :). If we can keep cloud traces requests and questions to official escalations it helps us have vision on the quality of our support and helps train folks across the board on how to support and operate Grafana Cloud Traces.

> IIUC none of your work here on Tempo is affecting this issue and instead its a deployment concern relegated only to Grafana Cloud?

correct

> The docs mention metric queries being experimental, is that something under your technical direction or another team?

yes. they will remain experimental until we are satisfied with the performance (just like original Tempo search did).

> And is the experimental aspect in anyway related to the 3h limit to your knowledge?

yes, but we recently rolled out a very large performance improvement which might allow us to extend this time period signficantly. we will use your escalated support request as an internal discussion point to see if the team is comfortable with raising the value.

**Reply by mdisibio** [reply]
Hi @jasonkuhrt,

What duration roughly would work for you?  Because metrics queries don't support streaming yet, we are limited in how far we can extend the range.  Without streaming a large query risks timing out and it doesn't provide feedback so it's impossible to tell if it's going to take 30s or minutes.   With streaming we will extend the range to the full extent like search.

**Reply by jasonkuhrt** [reply]
This is really helpful, thank you both. I don't want to abuse this forum so will raise that ticket. But to give a quick answer @mdisibio I _think_ 7 days would be a reasonable starting point for our team. That's a 56x increase over 3h haha so maybe I'm _way_ off what you were thinking (e.g. 2x, 5x).

**Reply by mdisibio** [reply]
We are on the same page.   I think 24h is feasible now, and 7-30d after we get streaming.

**Reply by jasonkuhrt** [reply]
@mdisibio That's great news. Could you indicate roughly when the streaming version would be available?

**Reply by icemanDD** [reply]
does the streaming stand for traceql stream: [Stream query results](https://grafana.com/docs/tempo/latest/traceql/#stream-query-results) or a traceql metrics dedicated implementation?

**Reply by mdisibio** [reply]
@jasonkuhrt Hoping for in the next couple of months.  There is remaining work both in Tempo and Grafana.  Oh also forgot to mention that Grafana Cloud Traces limit was increased to 24h.

@icemanDD It would be a new but similar implementation for metrics.
